include/oneapi/internal/execution_context_sycl.h

```diff
@@ -42,166 +42,175 @@ namespace interface1
 class OpenClKernelFactory : public Base, public ClKernelFactoryIface
 {
 private:
-    class ProgramCacheEntry
+    static void setStatus(services::Status * status = nullptr)
     {
-    private:
-        OpenClProgramRef * _program;
-
-    public:
-        ProgramCacheEntry() : _program(nullptr) {}
+        if (status != nullptr)
+            status->add(services::ErrorExecutionContext);
+    }
 
-        ~ProgramCacheEntry()
+    template<class T, size_t size>
+    class Map
+    {
+    private:
+        struct Entry
         {
-            delete _program;
-        }
+            services::SharedPtr<T> obj;
+            services::String name;
+        };
+        Entry* arr[size];
 
-        void setProgram(OpenClProgramRef *program)
+        size_t hash(const char* key)
         {
-            _program = program;
-        }
+            size_t hash    = 5381;
+            const char * str = key;
+            char c;
 
-        OpenClProgramRef * getProgram()
-        {
-            return _program;
+            while (c = *str++)
+            {
+                hash = ((hash << 5) + hash) + c; /* hash * 33 + c */
+            }
+            return hash;
         }
 
-        const char* getName(services::Status * status = nullptr)
+        size_t get_id(const char* name, services::Status * status = nullptr)
         {
-            if (!_program)
+            size_t id = hash(name) % size;
+            const size_t start_id = id;
+            while (arr[id] != nullptr && strcmp(arr[id]->name.c_str(), name) != 0)
             {
-                if (status)
+                id = (id + 1) % size;
+                if (id == start_id)
                 {
-                    status->add(services::ErrorExecutionContext);
+                    setStatus(status);
+                    return 0;
                 }
-                return nullptr;
             }
-            return _program->getName();
+            return id;
         }
-    };
-
-    class KernelCacheEntry
-    {
-    private:
-        KernelPtr _kernel;
-        services::String _name;
 
     public:
-        KernelCacheEntry() : _kernel(), _name() {}
+        Map()
+        {
+            for (size_t i = 0; i < size; ++i)
+                arr[i] = nullptr;
+        }
 
-        ~KernelCacheEntry() {}
+        ~Map()
+        {
+            for (size_t i = 0; i < size; ++i)
+            {
+                delete arr[i];
+                arr[i] = nullptr;
+            }
+        }
 
```

Comment:
This class does not implement neither copy/move constructor nor assignment operator. You have two options:
1. Implement :blush:
2. Prohibit it by moving it (C++03) to private section, StackOverflow shall help you